OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java

```diff
@@ -830,6 +830,9 @@ public RouteCalcResult searchRoute(final RoutingContext ctx, LatLon start, LatLo
 			HHRoutePlanner<NetworkDBPoint> routePlanner = HHRoutePlanner.create(ctx, null);
 			HHNetworkRouteRes r = null;
 			Double dir = ctx.config.initialDirection ;
+			if (targets.size() > 1) {
+				ctx.config.penaltyForReverseDirection /= 2; // relax reverse-penalty in case of inter-points
```

Comment:
1) not correct for 1st point 2) not applicable for other type of routings ?

Reply:
1. It was correct and now is clarified additionally.

3. Yes, this penalty is basically applied for inter-points routes for HH-engine only. If you decide to have the same for BRP-engine, I'll implement it.

Reply:
It's not correct, for the first point. It should be applied only to intermediates and revert later

Reply:
Fixed: 1) inter-points only, from second target; 2) reverted once after calculateHHRoute()